mrmap/registry/api/serializers/mapcontext.py

```diff
@@ -1,69 +1,175 @@
-from rest_framework.fields import SerializerMethodField
-from rest_framework.serializers import ModelSerializer, CharField
+from rest_framework.reverse import reverse
+from rest_framework.fields import SerializerMethodField, ModelField
+from rest_framework.serializers import CharField, HyperlinkedIdentityField
 
-from registry.api.serializers.service import OperationsUrlSerializer
-from registry.models import MapContext, MapContextLayer, OperationUrl
+from extras.api.serializers import NonNullModelSerializer
+from registry.models import MapContext, MapContextLayer
 
 
-class MapContextPropertiesSerializer(ModelSerializer):
-
-    class Meta:
-        model = MapContext
-        fields = '__all__'
-
-
-class MapContextLayerSerializer(ModelSerializer):
-    context_layer_operations = SerializerMethodField()
+class MapContextLayerPropertiesSerializer(NonNullModelSerializer):
+    title = ModelField(model_field=MapContextLayer()._meta.get_field('name'))
+    abstract = ModelField(model_field=MapContextLayer()._meta.get_field('title'))
+    offerings = SerializerMethodField()
+    folder = SerializerMethodField()
 
     class Meta:
         model = MapContextLayer
         fields = [
-            'id',
-            'parent',
-            'title',
-            # 'max_extent', # TODO
-            'context_layer_operations'
+            'title',  # 7.1.2.2 title
+            'abstract',  # 7.1.2.3 abstract
+            # TODO 'updated' # 7.1.2.4 updateDate
+            # TODO 'authors' # 7.1.2.5 author
+            # TODO 'publisher' # 7.1.2.6 publisher
+            # TODO 'rights' # 7.1.2.7 rights
+            # TODO 'date' # 7.1.2.9 temporalExtent
+            # TODO 'links' # 7.1.2.10 preview + 7.1.2.11 contentDescription + 7.1.2.12 contentByRef + 7.1.2.15 resourceMetadata
+            'offerings',  # 7.1.2.13 offering
+            # TODO 'active' # 7.1.2.14 active
+            # TODO 'categories' # 7.1.2.16 keyword
+            # TODO 'minScaleDenominator' # 7.1.2.17 minScaleDenominator
+            # TODO 'maxScaleDenominator' # 7.1.2.18 maxScaleDenominator
+            'folder',  # 7.1.2.19 folder
+            # TODO '*', # 7.1.2.20 extension
         ]
 
     @staticmethod
-    def get_max_extent(obj):
-        pass
-        # TODO
+    def __get_wms_offering(wms_layer):
+        return {
+            'code': 'http://www.opengis.net/spec/owc-geojson/1.0/req/wms',
+            'operations': [{
+                "code": "GetCapabilities",
+                "method": "GET",
+                "type": "application/xml",
+                "href": f"{wms_layer.service.url}service=WMS&request=GetCapabilities"
+            }, {
+                "code": "GetMap",
+                "method": "GET",
+                "type": "image/jpeg",  # TODO
+                "href": f"{wms_layer.service.url}service=WMS&version=1.3.0&request=GetMap&layer={wms_layer.identifier}"
```

Comment:
use 
```python
wms_layer.service.operation_urls.get(operation=OGCOperationEnum.GET_CAPABILITIES.value, method=HttpMethodEnum.GET.value).values_list('url', flat=True)
``` 
to get the correct url for the operations.

additionally you should not concatenate urls with strings. The url from db could looks like `https://host/bla?` or `https://host/bla?huhu=123&` it's not safe to concatenate urls. Instead use 
```python
import urllib.parse as urlparse
url = "http://www.example.com?type=a&type1=b&type2=c"
trigger = ["'or '1'='1'"," 'OR '1'='2'","'OR a=a"]

parsed = urlparse.urlparse(url)
querys = parsed.query.split("&")
result = []
for pairs in trigger:
    new_query = "&".join([ "{}{}".format(query, pairs) for query in querys])
    parsed = parsed._replace(query=new_query)
    result.append(urlparse.urlunparse(parsed))
```
to get a clean parsed url.